week3project/main.js

```diff
@@ -0,0 +1,81 @@
+const wrapper = document.querySelector(".wrapper"),
+inputPart = wrapper.querySelector(".input-part"),
+inputfield = inputPart.querySelector("input"),
+infotxt = inputPart.querySelector(".info-txt"),
+icons = document.querySelector(".weather-part img"),
+backarrow = wrapper.querySelector(".h1 i"),
+searchhistory = wrapper.querySelector(".searches");
+let api;
+
+
+backarrow.addEventListener("click", function(){
+    wrapper.classList.remove("active");
+});
+
+inputfield.addEventListener("keyup", function(e){
+if(e.key === "Enter" && inputfield.value != ""){
+    requestApi(inputfield.value);       
+}
+setTimeout(() => {
+  if(e.key === "Enter" && inputfield.value != ""){
+       var paragraph = document.createElement("p");
+       paragraph.innerText = inputfield.value;
+        searchhistory.appendChild(paragraph);
+        
+    }
+ }, 3000);
+});
```

Comment:
The problem with using this approach is - 
1. What if the city is invalid? it still gets added to the history.
2. What if the request to the API takes > 3 seconds?